### ISSUE TITLE
Removing file with "problems" from outside of VSCode not recognized 

### ISSUE THREAD

**Issue by gus** [microsoft/pyright#160]
**Describe the bug**
When pyright is reporting problems for a file and that file is then removed from the file-system via a mechanism external to VSCode (i.e. a terminal), pyright continues to report problems for the removed file. This is not an issue when removing the file within VSCode.

**To Reproduce**
1. Create a file named `foo.py` with the following contents:

```python
x: dict = {}
y: int = x.get("foo")
```

Notice problems reported for `foo.py` in the Problems tab.

2. In a terminal, even the one provided by VSCode, run `rm foo.py`. Notice that problems are still reported for `foo.py`.

**Expected behavior**
After file removal, I expect `foo.py` not have any listings under the Problems tab.

**VS Code extension or command-line**
VSCode: `v1.35.1` 
pyright extension: `v1.0.41`
Linux: `v4.18.0-20-generic`.

**Additional context**
This is not an issue related to too many files being watched.

Cheers,
Gus

**Comment by gus** [comment]
Well, going to close this for now as I can't reliably recreate the issue at the moment. Though prior to reporting the bug, when I closed the window for the deleted file, `foo.py` was still reported as having issues in the Problems tab, since reloading my vscode instance I can not recreate the problem.

Which leads me to think this was not an issue with pyright and more of an issue with vscode itself.

Sorry for the troubles.

Cheers,
Gus

**Comment by erictraut** [comment]
I've seen this before as well, but it's not consistent. I suspect that it's either a bug in the file system watcher that's built into node (it has historically been relatively unreliable and buggy) or there's a timing bug in the pyright code. I'm going to re-open the bug and investigate further. Thanks for reporting.

**Comment by erictraut** [comment]
I found and fixed a bug related to the pyrightconfig.json config file. The file system watcher was not properly handling the case where the config file was deleted and re-created, which can happen when switching branches in git. But I wasn't able to repro any problems with the file system watchers used to detect changes in source files. It's possible that this problem is specific to Linux. I made one change here to make it slightly more robust. I also added a bit more logging to help further diagnose this problem. This will be in the next version of pyright. If you continue to see this problem after you receive the next update, let me know, and I'll investigate further.
